assets/app/scripts/controllers/deployments.js

```diff
@@ -85,6 +120,222 @@ angular.module('openshiftConsole')
+      DataService.create("deploymentconfigrollbacks", null, req, $scope).then(
+        function(newDeploymentConfig) {
+          // update the deployment config based on the one returned by the rollback
+          DataService.update("deploymentconfigs", deploymentConfigName, newDeploymentConfig, $scope).then(
+            function(rolledBackDeploymentConfig) {
+                $scope.alerts = [
+                {
+                  type: "success",
+                  message: "Deployment #" + rolledBackDeploymentConfig.status.latestVersion + " is rolling back " + deploymentConfigName + " to " + deploymentName + ".",
+                }
+              ];
+            },
+            function(result) {
+              $scope.alerts = [
+                {
+                  type: "error",
+                  message: "An error occurred while rolling back the deployment.",
+                  details: getErrorDetails(result)
+                }
+              ];
+            }
+          );
+        },
+        function(result) {
+          $scope.alerts = [
+            {
+              type: "error",
+              message: "An error occurred while rolling back the deployment.",
+              details: getErrorDetails(result)
+            }
+          ];
+        }
+      );
+    };
+
+    $scope.cancelRunningDeployment = function(deploymentConfigName, deploymentName) {
+      var deployment = $scope.deploymentsByDeploymentConfig[deploymentConfigName][deploymentName];
+      var req = deployment;
+
+      // set the cancellation annotations
+      var deploymentCancelledAnnotation = $filter('annotationName')('deploymentCancelled');
+      var deploymentStatusReasonAnnotation = $filter('annotationName')('deploymentStatusReason');
+      req.metadata.annotations[deploymentCancelledAnnotation] = "true";
+      req.metadata.annotations[deploymentStatusReasonAnnotation] = "The deployment was cancelled by the user";
+
+      // update the deployment with cancellation annotations
+      DataService.update("replicationcontrollers", deploymentName, req, $scope).then(
+        function() {
+            $scope.alerts = [
+            {
+              type: "success",
+              message: "Retrying deployment " + deploymentName + " of " + deploymentConfigName + ".",
+            }
+          ];
+        },
+        function(result) {
+          $scope.alerts = [
+            {
+              type: "error",
+              message: "An error occurred while cancelling the deployment.",
+              details: getErrorDetails(result)
+            }
+          ];
+        }
+      );
+    };
+
+    $scope.deploymentIsLatest = function(deploymentConfig, deployment) {
+      var deploymentVersion = parseInt($filter('annotation')(deployment, 'deploymentVersion'));
+      var deploymentConfigVersion = deploymentConfig.status.latestVersion;
+      return deploymentVersion === deploymentConfigVersion;
+    };
+
+    $scope.deploymentStatus = function(deployment) {
```

Comment:
actually i will make an exception in this case given how much you are having to use it :)